main.tf

```
@@ -54,12 +54,29 @@ locals {
   #   "../root-module-a/stack.yaml",
   #   "../root-module-b/stack.yaml",
   # ]
-  _multi_instance_stack_files  = fileset("${path.root}/${var.root_modules_path}/*/stacks", "*.yaml")
-  _single_instance_stack_files = fileset("${path.root}/${var.root_modules_path}/*", "stack.yaml")
+  # This includes nested directories, example: [
+  #   "ecs-infrastructure/service-1/stacks/example.yaml",
+  #   "ecs-infrastructure/service-1/stacks/common.yaml",
+  #   "ecs-infrastructure/service-2/stacks/example.yaml",
+  #   "ecs-infrastructure/service-2/stacks/common.yaml",
+  # ]
+  _multi_instance_stack_files_raw  = fileset("${path.root}/${var.root_modules_path}", "**/stacks/*.yaml")
+  _single_instance_stack_files_raw = fileset("${path.root}/${var.root_modules_path}", "**/stack.yaml")
+
+  # Filter out any files that are in .terraform directories to avoid picking up module cache now that it's using ** as the wildcard
+  _multi_instance_stack_files  = [for file in local._multi_instance_stack_files_raw : file if !can(regex("\\.terraform/", file))]
+  _single_instance_stack_files = [for file in local._single_instance_stack_files_raw : file if !can(regex("\\.terraform/", file))]
   _all_stack_files             = local._multi_instance_structure ? local._multi_instance_stack_files : local._single_instance_stack_files
 
   # Extract the root module name from the stack file path
-  _all_root_modules = distinct([for file in local._all_stack_files : dirname(replace(replace(file, "../", ""), "stacks/", ""))])
+  # For MultiInstance: extract path before "/stacks" to get the full nested path
+  # For SingleInstance: extract directory path to get the full nested path
+  _all_root_modules = distinct([
+    for file in local._all_stack_files :
+    local._multi_instance_structure ?
+    dirname(dirname(file)) : # For MultiInstance: example2/nested/stacks/stack.yaml -> example2/nested
```

```suggestion
    replace(file, "/stacks/.*$", "") : # For MultiInstance: example2/nested/stacks/stack.yaml -> example2/nested
```
I suggest supporting any number of nested dirs. Something like this should do the trick.

This removes everything from `/stacks/` onward in the path and should work for any nesting level, like:
`example2/nested/deeply/stacks/stack.yaml` → `example2/nested/deeply`